### ISSUE TITLE
Bio::Factory::EMBOSS dies on OSX when checking for operating system

### ISSUE THREAD

**Issue by piatekmj** [bioperl/bioperl-run#46]
This is in reference to a [legacy ticket](https://rt.cpan.org/Public/Bug/Display.html?id=73061) I opened in 2011 that has not been resolved yet.

The same issue was encountered with the below session info:
perl = v5.22.0
Mac OS = 10.11.6
BioPerl-Run-1.007001

I was able to overcome this issue only by installing EMBOSS through an anaconda package manager for Mac OS. I was not able to resolve this issue via any other alternative way.

After installing EMBOSS via anaconda, I was able to successfully execute the code quoted in the legacy ticket.

**Comment by bosborne** [comment]
Marek,

I’m not able to reproduce this, I can run the test script attached to that legacy ticket. But since you’ve said that you are able to execute the script as well it’s not clear to me what should be done here.

Are you saying that when you install EMBOSS by some method other than anaconda you can not run the test script? Or, are you saying that when EMBOSS is not installed you can not run the test script?

Brian O.

**Comment by piatekmj** [comment]
Apologies if I was not clear.

I would like to be able to run and control EMBOSS via BioPerl in a script. I have done that in the past on other OS.

On Mac, when I installed Bio::Factory::EMBOSS via cpanm, I had to force it otherwise it failed. This method got me stuck (5 years ago and also now) at the point reported in the legacy ticket. So I was not able to print the second 'check' from that script.

After installing EMBOSS with anaconda package manager, somehow my BioPerl script reported in the legacy ticket started _working_ and I managed to see the second 'check' printed. There, however, have been other problems, perhaps related to this that I reported in another ticket.

To sum up with regards to your question:
I am unable to run my test script if I install EMBOSS by method other than anaconda.
However I do not intend to run EMBOSS straight from command line. I would like to have EMBOSS working in a script.

Thanks,
Marek

PS.
The legacy script for broader audience is pasted below:
```
#!/usr/bin/perl -w
use strict;
use Bio::Factory::EMBOSS;
$|++;
my $factory = Bio::Factory::EMBOSS -> new();
my $prog = $factory->program('water');
my %input = (
 -asequence => 'asis:gatcgatcgatgcat',
 -bsequence => 'asis:cgatgcatgctactg',
);
print "CHECK 1\n";
$prog->run(\%input);
print "CHECK 2\n";
```

**Comment by bosborne** [comment]
Marek,

I see. So you are still having problems, as reported in that other email.

So as I asked earlier, please show us the code. If I understand this correctly the code below works, so we need to see the code that does not work.

Brian O.

**Comment by piatekmj** [comment]
Brian,

Please refer to ticket #47 for more detailed description of the problem.

Although I was able to overcome the problem with this simple script, I think it still remains a mystery why I was not able to do that with installing EMBOSS via cpan only (using exactly the same code).

Ultimately, this ticket is intended to help sort out issues with regards to OS, where checking for Mac OS stalls execution of the script.

It seems that since I was able to overcome the system check problem, this ticket turns into more informational now.

Please close the ticket if you think there is nothing to be done.

Thanks,
Marek

**Comment by bosborne** [comment]
Closing this, at Marek's suggestion.
